barretenberg/cpp/src/barretenberg/vm/avm_trace/avm_trace.cpp

```diff
@@ -1179,6 +1180,199 @@ void AvmTraceBuilder::op_timestamp(uint32_t dst_offset)
     main_trace.push_back(row);
 }
 
+// Helper function to add kernel lookup operations into the main trace
+Row AvmTraceBuilder::create_kernel_output_opcode(uint32_t clk, uint32_t data_offset, AvmMemoryTag r_tag)
```

Comment:
We always call this method with FF type for r_tag. Maybe we can remove it form the arguments?
Same for the two other routines.